packages/authentication/android/src/main/java/dev/robingenz/capacitorjs/plugins/firebase/authentication/handlers/GoogleAuthProviderHandler.java

```diff
@@ -31,6 +37,25 @@ public GoogleAuthProviderHandler(FirebaseAuthentication pluginImplementation) {
     }
 
     public void signIn(PluginCall call) {
+        JSArray scopes = call.getArray("scopes");
+        // Got scopes? Need to get the client again with these new options.
+        if (scopes != null) {
+            try {
+                List<String> scopeList = scopes.toList();
+                GoogleSignInOptions.Builder gsob = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                    .requestIdToken(pluginImplementation.getPlugin().getContext().getString(R.string.default_web_client_id))
+                    .requestEmail();
+
+                for (String scope : scopeList) {
+                    gsob = gsob.requestScopes(new Scope(scope));
+                }
+
+                mGoogleSignInClient = GoogleSignIn.getClient(pluginImplementation.getPlugin().getActivity(), gsob.build());
+            } catch (JSONException exception) {
+                Log.e(FirebaseAuthenticationPlugin.TAG, "signIn (applying scopes) failed.", exception);
+            }
+        }
```

Comment:
Please move this code to a new method, e.g.:

```java
mGoogleSignInClient = buildGoogleSignInClient(scopeList);
```

This method can then also be called from the constructor with an empty list.